Add all docs to the website (#235)
kubernetes-sigs/kernel-module-management: docs/mkdocs/documentation/deploy_kmod.md

## `Module` CRD

The `Module` Custom Resource Definition represents a kernel module that should be loaded on all or select nodes in the
cluster, through a ModuleLoader image.
A Module specifies one or more kernel versions it is compatible, as well as a node selector.

The compatible versions for a `Module` are listed under `.spec.moduleLoader.container.kernelMappings`.
A kernel mapping can either match a `literal` version, or use `regexp` to match many of them at the same time.

The reconciliation loop for `Module` runs the following steps:

- list all nodes matching `.spec.selector`;
- build a set of all kernel versions running on those nodes;
- for each kernel version:
  - go through `.spec.moduleLoader.container.kernelMappings` and find the appropriate container image name.
    If the kernel mapping has `build` or `sign` defined and the container image does not already exist, run the build
    and / or signing job as required;
  - create a ModuleLoader `DaemonSet` with the container image determined at the previous step;
  - if `.spec.devicePlugin` is defined, create a device plugin `DaemonSet` using the configuration specified under
    `.spec.devicePlugin.container`;

> - garbage-collect existing `DaemonSets` targeting kernel versions that are not run by any node in the cluster.

Reviewer: we should also add:
- garbage collect successfully finished build jobs
- garbage collect successfully finished sign jobs

metadata:
  name: my-kmod
spec:
  moduleLoader:
    container:
      modprobe:
        moduleName: my-kmod  # Required

        dirName: /opt  # Optional

        # Optional. Will copy /firmware/* into /var/lib/firmware/ on the node.
        firmwarePath: /firmware
        
        parameters:  # Optional
          - param=1

      kernelMappings:  # At least one item is required
        - literal: 6.0.15-300.fc37.x86_64
          containerImage: some.registry/org/my-kmod:6.0.15-300.fc37.x86_64

        # For each node running a kernel matching the regexp below,
        # KMM will create a DaemonSet running the image specified in containerImage
        # with ${KERNEL_FULL_VERSION} replaced with the kernel version.
        - regexp: '^.+\fc37\.x86_64$'
          containerImage: "some.other.registry/org/my-kmod:${KERNEL_FULL_VERSION}"

        # For any other kernel, build the image using the Dockerfile in the my-kmod ConfigMap.
        - regexp: '^.+$'
          containerImage: "some.registry/org/my-kmod:${KERNEL_FULL_VERSION}"
          build:
            buildArgs:  # Optional
              - name: ARG_NAME
                value: some-value
            secrets:  # Optional

>               - name: some-kubernetes-secret  # Will be mounted in the build pod as /run/secrets/some-kubernetes-secret.

Reviewer 2: ```suggestion
              - name: some-kubernetes-secret  # Will be available to the dockerfile at /run/secrets/some-kubernetes-secret.
```

The fact that kaniko parse the dockerfile in its own process is an implementation detail (so /run/secrets in the build pod and /run/secrets in the Dockerfile is the same).

In addition to that, it won't be the case in OCP, the build pod will contains the secrets in a completely different path and mounted to the build env at `/run/secrets/...`

Author: WDYT about
```yaml
# Will be available in the build environment at /run/secrets/some-kubernetes-secret.
```

Reviewer 2: I am OK with it. As long as we remove the "build pod" stuff.
